### PR TITLE
Find Usages for LSP languages - both client and (Java) server.

### PR DESCRIPTION
**Pull request by jlahoda**

### CODE REVIEW

File: java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/text/TextDocumentServiceImpl.java [hunk 1]
```diff
@@ -591,16 +588,69 @@ public static String html2MD(String html) {
...
+                    cc.toPhase(JavaSource.Phase.RESOLVED);
+                    if (cancel.get()) return ;
+                    Document doc = cc.getSnapshot().getSource().getDocument(true);
+                    TreePath path = cc.getTreeUtilities().pathFor(getOffset(doc, params.getPosition()));
+                    query[0] = new WhereUsedQuery(Lookups.singleton(TreePathHandle.create(path, cc)));
+                }, true);
+                if (cancel.get()) return ;
+                cancelCallback[0] = () -> query[0].cancelRequest();
+                RefactoringSession refactoring = RefactoringSession.create("FindUsages");
+                query[0].fastCheckParameters();
```
Comment (matthiasblaesing) [inline java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/text/TextDocumentServiceImpl.java:622]: I'm not familiar with the refactoring API, but my reading of the Javadoc is, that:

- preCheck
- prepare
- checkParameters
- fastCheckParameters

do not raise exceptions, but return "Problem" instances. The way they are invoked here, the return value is ignored. Not sure whether this is intenential, but it looks wrong.

My reading of 

http://bits.netbeans.org/dev/javadoc/org-netbeans-modules-refactoring-api/org/netbeans/modules/refactoring/api/AbstractRefactoring.html#fastCheckParameters--

is, that the call to `checkParameters` should do a superset of checks to `fastCheckParameters`. So if `checkParameters` is invoked, I would expect the call to `fastCheckParameters` to be superfluous.

**Review by matthiasblaesing — COMMENTED** [review]
For the server part I left an inline comment. For the client part I tested with typescript and worked good. The only missing thing is integration into the context menu. If that is a known limitation right now, all good, if not, I said it now :-)

**Comment by jlahoda** [comment]
Thanks for the comments, @matthiasblaesing . I tried to add the action to the editor popup, and added some handling of the refactoring Problems - basically fatal problems lead to a reported exception sent to the client.

**Review by matthiasblaesing — APPROVED** [review]
Thank you, looks sane to me - let's get this in.

File: ide/lsp.client/nbproject/project.properties [hunk 2]
```diff
@@ -18,10 +18,10 @@
 javac.source=1.8
 javac.compilerargs=-Xlint -Xlint:-serial
 javadoc.arch=${basedir}/arch.xml
-release.external/org.eclipse.lsp4j-0.8.1-p1.jar=modules/ext/org.eclipse.lsp4j-0.8.1-p1.jar
-release.external/org.eclipse.lsp4j.generator-0.8.1.jar=modules/ext/org.eclipse.lsp4j.generator-0.8.1.jar
-release.external/org.eclipse.lsp4j.jsonrpc-0.8.1.jar=modules/ext/org.eclipse.lsp4j.jsonrpc-0.8.1.jar
-release.external/org.eclipse.xtend.lib-2.18.0.jar=modules/ext/org.eclipse.xtend.lib-2.18.0.jar
-release.external/org.eclipse.xtend.lib.macro-2.18.0.jar=modules/ext/org.eclipse.xtend.lib.macro-2.18.0.jar
-release.external/org.eclipse.xtext.xbase.lib-2.18.0.jar=modules/ext/org.eclipse.xtext.xbase.lib-2.18.0.jar
+release.external/org.eclipse.lsp4j-0.9.0.jar=modules/ext/org.eclipse.lsp4j-0.9.0.jar
```
Comment (JaroslavTulach) [inline ide/lsp.client/nbproject/project.properties:21]: There are similar changes on my [java LSP debugging branch](https://github.com/JaroslavTulach/netbeans/tree/java-lsp-server-debugging-attempt2). Feel free to integrate soon. I will resolve the merge conflicts next week then.

**Review by JaroslavTulach — APPROVED** [review]

**Comment by matthiasblaesing** [comment]
@jlahoda any reason not to merge this?

**Comment by jlahoda** [comment]
Given the overall situation, lets wait for https://github.com/apache/netbeans/pull/2447, then I'll resolve the conflicts and merge into master.

**Comment by lkishalmi** [comment]
#2447 has been merged, let's resolve the conflicts!

**Comment by jlahoda** [comment]
Done.

**Comment by arvindaprameya** [comment]
Approved, thanks for doing this Jan

**Pull request merged**